org-journal.el

```diff
@@ -381,6 +381,173 @@ If the date is not today, it won't be given a time."
     (org-journal-mark-entries)
     (calendar-exit)))
 
+;;; Journal search facilities
+;;
+
+(defun org-journal-search (str &optional period-name)
+  "Search for a string in the journal within a given interval.
+See `org-read-date` for information on ways to specify dates."
+  (interactive (list (read-string "Enter a string to search for: " nil 'org-journal-search-history)))
+  (let* ((period-pair (org-journal-read-period period-name))
```

Comment:
At this point, I think that a prefix argument that switches from a bounded search to a search across all files is really useful:

```
(let* ((period-pair (org-journal-read-period (if current-prefix-arg 'forever period-name)))
...
```